### ISSUE TITLE
[Bug]: Attempting to select non-boolean expression

### ISSUE THREAD

**Issue by chrysanthemum-boy** [infiniflow/infinity#683]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

11fe044a

### Actual behavior

```
[critical] Attempting to select non-boolean expression@src/executor/expression/expression_selector.cpp:69
terminate called after throwing an instance of 'infinity::UnrecoverableException@infinity_exception'
  what():  Attempting to select non-boolean expression@src/executor/expression/expression_selector.cpp:69
Aborted

```

### Steps to reproduce

```Markdown
Function test code can be found in python/test/test_convert.py`test_output_filter_function`
Please use "c1".
```

**Comment by JinHai-CN** [comment]
Fixed by #690

**Comment by chrysanthemum-boy** [comment]
It looks fixed.
